pkg/dc2deployment/dc2deployment.go

```diff
@@ -96,13 +87,63 @@ func Mutate(pluginName string, dc dcAPI.DeploymentConfig, log *logrus.Logger) (d
 			deploy.Spec.Selector.MatchLabels[index] = element
 		}
 	}
-	log.Debugf("[%s] d.Selector: %#v", pluginName, deploy.Spec.Selector)
+	log.Debugf("[%s] Strategy: %#v", pluginName, dc.Spec.Strategy)
+
+	// openshift supoorts Strategy.Type={"Recreate","Custom","Rolling"}
+	// K8s native supports k8s.Strategy.Type={"Recreate","RollingUpdate"}
+	// Custom strategy.type is not supported in K8s native, hence defaulting it to RollingUpdate
+	if dc.Spec.Strategy.Type != "" {
+		if dc.Spec.Strategy.Type == "Rolling" || dc.Spec.Strategy.Type == "Custom" {
+			deploy.Spec.Strategy.Type = deployAPI.DeploymentStrategyType("RollingUpdate")
+		} else {
+			deploy.Spec.Strategy.Type = deployAPI.DeploymentStrategyType(dc.Spec.Strategy.Type)
+		}
+
+	}
+	//Rollingupdate.MaxSurge and Rollingupdate.MaxUnavailable supported only when Strategy type is "RollingUpdate"
+	//If openshift Strategy.type is Custom Rollingupdate.MaxSurge and Rollingupdate.MaxUnavailable is not applicable.
+	if deploy.Spec.Strategy.Type == "RollingUpdate" && dc.Spec.Strategy.Type != "Custom" {
+		deploy.Spec.Strategy.RollingUpdate = new(deployAPI.RollingUpdateDeployment)
+		deploy.Spec.Strategy.RollingUpdate.MaxSurge = dc.Spec.Strategy.RollingParams.MaxSurge
+		deploy.Spec.Strategy.RollingUpdate.MaxUnavailable = dc.Spec.Strategy.RollingParams.MaxUnavailable
+	}
+	deploy.Spec.RevisionHistoryLimit = dc.Spec.RevisionHistoryLimit
 
 	if dc.Spec.Template != nil {
 		dc.Spec.Template.DeepCopyInto(&deploy.Spec.Template)
 	}
-
-	log.Tracef("[%s] deployment %#v", pluginName, deploy)
-
+	// End of Spec Section
+
+	//Status Section Start
```

Comment:
We do not need any conversion on the Status field, please drop all the conversion logic. Reason: The Status field is populated by the controller responsible for the object. In this case, we are creating a brand-new object and we have to let the controller handle those. Populating the Status from a different object running on a cluster is misleading.

Reply:
Removed the status filed assignment.